indico/modules/events/registration/util_test.py

```diff
@@ -418,3 +422,181 @@ def test_get_registered_event_persons(dummy_event, dummy_user, dummy_regform):
+        ]
+    })
+    db.session.add(multi_choice_field)
+    db.session.flush()
+
+    data = {
+        boolean_field.html_field_name: True,
+        multi_choice_field.html_field_name: {'AbntJPo-cMPo8SKgcsjAr': 2},
+        'email': dummy_user.email, 'first_name': dummy_user.first_name, 'last_name': dummy_user.last_name}
+    reg = create_registration(dummy_regform, data, invitation=None, management=False, notify_user=False)
+
+    assert reg.data_by_field[boolean_field.id].data
+    assert reg.data_by_field[multi_choice_field.id].data == {'AbntJPo-cMPo8SKgcsjAr': 2}
+    db.session.delete(reg)
+    db.session.flush()
+
+    # Make sure that missing data gets default values:
+    data = {
+        'email': dummy_user.email, 'first_name': dummy_user.first_name, 'last_name': dummy_user.last_name}
+    reg = create_registration(dummy_regform, data, invitation=None, management=False, notify_user=False)
+
+    assert not reg.data_by_field[boolean_field.id].data
+    assert reg.data_by_field[multi_choice_field.id].data == {}
+    db.session.delete(reg)
+    db.session.flush()
+
+    # Add a manager only section
+    section = RegistrationFormSection(registration_form=dummy_regform, title='manager_section', is_manager_only=True)
+    db.session.add(section)
+    db.session.flush()
+
+    checkbox_field = RegistrationFormField(parent_id=section.id, registration_form=dummy_regform)
+    _fill_form_field_with_data(checkbox_field, {
+        'input_type': 'checkbox', 'title': 'Checkbox'
+    })
+    db.session.add(checkbox_field)
+    db.session.flush()
+
+    data = {
+        checkbox_field.html_field_name: True,
+        'email': dummy_user.email, 'first_name': dummy_user.first_name, 'last_name': dummy_user.last_name}
+    reg = create_registration(dummy_regform, data, invitation=None, management=False, notify_user=False)
+
+    assert not reg.data_by_field[boolean_field.id].data
+    assert reg.data_by_field[multi_choice_field.id].data == {}
+    # Assert that the manager field gets the default value, not the value sent
+    assert not reg.data_by_field[checkbox_field.id].data
+    db.session.delete(reg)
+    db.session.flush()
+
+    # Try again with management=True
+    data = {
+        checkbox_field.html_field_name: True,
+        'email': dummy_user.email, 'first_name': dummy_user.first_name, 'last_name': dummy_user.last_name}
+    reg = create_registration(dummy_regform, data, invitation=None, management=True, notify_user=False)
+
+    assert not reg.data_by_field[boolean_field.id].data
+    assert reg.data_by_field[multi_choice_field.id].data == {}
+    # Assert that the manager field gets properly set with management=True
+    assert reg.data_by_field[checkbox_field.id].data
+
+
+def test_modify_registration(monkeypatch, dummy_event, dummy_user, dummy_regform):
+    class _DummySesssion:
+        user = dummy_user
+    monkeypatch.setattr('indico.modules.events.registration.util.session', _DummySesssion)
```

Comment:
You could also use the `request_context` fixture (using `@pytest.mark.usefixtures('request_context')`) and then call `session.set_session_user(dummy_user)`